repo/object/object_manager.go

```diff
@@ -86,7 +86,7 @@ func (om *Manager) NewWriter(ctx context.Context, opt WriterOptions) Writer {
 	}
 
 	w.buffer.Reset()
-	w.contentWriteError = nil
+	w.setError(nil)
```

Comment:
It appears that resetting `contentWriteError` under `w.contentWriteErrorMutex` is not strictly necessary, given that at this point there _"should be no other references"_ to `w`. 🤷🏼